### PR TITLE
Use mamba to build the environment instead of conda

### PR DESCRIPTION
**Pull request by ewels**
Tested locally, seems to work fine (and quite quickly).

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/diaproteomics/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/diaproteomics _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,19 +2,22 @@ FROM nfcore/base:1.14
...
+# Install mamba
+RUN conda install mamba -n base -c conda-forge
+
 # Install the conda environment
 COPY environment.yml /
-RUN conda env create --quiet -f /environment.yml && conda clean -a
+RUN mamba env create --quiet -f /environment.yml && mamba clean -a
 
 # Add conda installation dir to PATH (instead of doing 'conda activate')
 ENV PATH /opt/conda/envs/nf-core-diaproteomics-1.2.5dev/bin:$PATH
```
Comment (Leon-Bichmann) [inline Dockerfile:13]: does the path need to be changed to mamba as well?

Comment (Leon-Bichmann) [inline Dockerfile:13]:
```suggestion
ENV PATH /opt/mamba/envs/nf-core-diaproteomics-1.2.5dev/bin:$PATH
```

Comment (ewels) [reply Dockerfile:13]: I wondered that but I don't think so.

Just checked with `docker run -it nfcore/diaproteomics:dev sh` and the path seems fine 👍🏻

Comment (Leon-Bichmann) [reply Dockerfile:13]: just tested it and had a problem with running `pyprophet --version` due to a missing library

**Review by Leon-Bichmann — APPROVED** [review]

**Pull request merged**